lib/routes/withdraw/reverse_swap/confirmation_page/reverse_swap_confirmation_page.dart

```diff
@@ -0,0 +1,118 @@
+import 'package:breez_translations/breez_translations_locales.dart';
+import 'package:c_breez/bloc/withdraw/withdraw_funds_bloc.dart';
+import 'package:c_breez/bloc/withdraw/withdraw_funds_state.dart';
+import 'package:c_breez/routes/withdraw/reverse_swap/confirmation_page/widgets/reverse_swap_button.dart';
+import 'package:c_breez/routes/withdraw/widgets/fee_chooser/fee_chooser.dart';
+import 'package:c_breez/widgets/loader.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class ReverseSwapConfirmationPage extends StatefulWidget {
+  final int amountSat;
+  final String onchainRecipientAddress;
+  final String feesHash;
+  final int? boltzFees;
+
+  const ReverseSwapConfirmationPage({
+    Key? key,
+    required this.amountSat,
+    required this.onchainRecipientAddress,
+    required this.feesHash,
+    this.boltzFees,
+  }) : super(key: key);
+
+  @override
+  State<ReverseSwapConfirmationPage> createState() => _ReverseSwapConfirmationPageState();
+}
+
+class _ReverseSwapConfirmationPageState extends State<ReverseSwapConfirmationPage> {
+  List<FeeOption> affordableFees = [];
+  int selectedFeeIndex = -1;
+
+  late Future<List<FeeOption>> _fetchFeeOptionsFuture;
+
+  @override
+  void initState() {
+    super.initState();
+    _fetchFeeOptionsFuture = context.read<WithdrawFundsBloc>().fetchFeeOptions();
+    _fetchFeeOptionsFuture.then((feeOptions) {
+      setState(() {
+        affordableFees = feeOptions.where((f) => f.isAffordable(widget.amountSat)).toList();
+        selectedFeeIndex = (affordableFees.length / 2).floor();
+      });
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final texts = context.texts();
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(texts.sweep_all_coins_speed),
+      ),
+      body: FutureBuilder(
+        future: _fetchFeeOptionsFuture,
+        builder: (context, snapshot) {
+          if (snapshot.error != null) {
+            return _ErrorMessage(
```

Comment:
What happens here? Do we allow to retry with a button?

Reply:
No, user has to go to the previous page and retry again.
This is the behavior on breezmobile but it can be improved as you suggest.

Reply:
Let's leave it as in breez mobile.